controllers/operators/operators_objects.go

```diff
@@ -0,0 +1,290 @@
+	s.Spec = &ov1alpha1.SubscriptionSpec{
+		CatalogSource:          "community-operators",
+		CatalogSourceNamespace: "openshift-marketplace",
+		Package:                "grafana-operator",
+		Channel:                "v4",
+		StartingCSV:            "grafana-operator.v4.2.0",
+		InstallPlanApproval:    "Automatic",
+	}
+	return s
+}
+
+func strimziSubscription(name string, namespace string) *ov1alpha1.Subscription {
+	s := subscription(name, namespace)
+	s.Spec = &ov1alpha1.SubscriptionSpec{
+		CatalogSource:          "community-operators",
+		CatalogSourceNamespace: "openshift-marketplace",
+		Package:                "strimzi-kafka-operator",
+		Channel:                "stable",
+		StartingCSV:            "strimzi-cluster-operator.v0.28.0",
+		InstallPlanApproval:    "Automatic",
+	}
+	return s
+}
+
+func lokiSubscription(name string, namespace string) *ov1alpha1.Subscription {
+	s := subscription(name, namespace)
+	s.Spec = &ov1alpha1.SubscriptionSpec{
+		CatalogSource:          constants.ObservabilityName,
+		CatalogSourceNamespace: namespace,
+		Package:                "loki-operator",
+		Channel:                "tech-preview",
+		StartingCSV:            "loki-operator.v0.0.1",
+		InstallPlanApproval:    "Automatic",
+	}
+	return s
+}
+
+func prometheusSubscription(name string, namespace string) *ov1alpha1.Subscription {
+	s := subscription(name, namespace)
+	s.Spec = &ov1alpha1.SubscriptionSpec{
+		CatalogSource:          "community-operators",
+		CatalogSourceNamespace: "openshift-marketplace",
+		Package:                "prometheus",
+		Channel:                "beta",
+		StartingCSV:            "prometheusoperator.0.47.0",
+		InstallPlanApproval:    "Automatic",
+	}
+	return s
+}
+
+func kafkaInstance(ns string, spec *v1alpha1.KafkaInstanceSpec) *sv1b2.Kafka {
+	return &sv1b2.Kafka{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Kafka",
+			APIVersion: "kafka.strimzi.io/v1beta2",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      constants.KafkaName,
+			Namespace: ns,
+			Labels:    map[string]string{"part-of": constants.ObservabilityName},
+		},
+		Spec: &sv1b2.KafkaSpec{
+			Kafka: sv1b2.KafkaSpecKafka{
+				Config:    spec.Kafka.Config,
+				Listeners: spec.Kafka.Listeners,
+				Replicas:  spec.Kafka.Replicas,
+				Storage:   spec.Kafka.Storage,
+				Version:   spec.Kafka.Version,
+			},
+			Zookeeper: sv1b2.KafkaSpecZookeeper{
+				Replicas: spec.Zookeeper.Replicas,
+				Storage:  spec.Zookeeper.Storage,
+			},
+		},
+	}
+}
+
+func kafkaTopic(ns string, spec *v1alpha1.FlowCollectorKafka) *sv1b2.KafkaTopic {
+	return &sv1b2.KafkaTopic{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "KafkaTopic",
+			APIVersion: "kafka.strimzi.io/v1beta2",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      spec.TopicName, // kafka topic name is set from metadata name
+			Namespace: ns,
+			Labels:    map[string]string{"part-of": constants.ObservabilityName},
+		},
+		Spec: &sv1b2.KafkaTopicSpec{
+			Config:     spec.TopicSpec.Config,
```

Comment:
getting segfault here .. I'm using the provided flowcollector_dev yaml, but it doesn't have a `TopicSpec` configured. I guess it doesn't have to be a mandatory configuration, does it?

Here's my logs:

```
1.6468341854308603e+09	INFO	controller.flowcollector	subscription strimzi-kafka-operator already exists	{"reconciler group": "flows.netobserv.io", "reconciler kind": "FlowCollector", "name": "cluster", "namespace": "", "component": "Operators"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x14ddb40]

goroutine 454 [running]:
github.com/netobserv/network-observability-operator/controllers/operators.kafkaTopic(...)
	/opt/app-root/controllers/operators/operators_objects.go:165
github.com/netobserv/network-observability-operator/controllers/operators.(*Reconciler).getInstances(0xc000ae5b88, {0x18f711d, 0x16})
	/opt/app-root/controllers/operators/operators_reconciler.go:202 +0x1400
github.com/netobserv/network-observability-operator/controllers/operators.(*Reconciler).manageInstance(0xc000ae5b88, {0x1b015b8, 0xc001db0a20}, {0x18f711d, 0x16})
	/opt/app-root/controllers/operators/operators_reconciler.go:165 +0x5c

```

Reply:
good catch :+1: nothing should be mandatory. I will add some logic there
It seems there is a similar case for grafana `DashboardSpec`